cmsis-core/cmsis_nvic.h

```diff
@@ -34,6 +34,12 @@
 
 #include "cmsis.h"
 
+/* throw meaningful compiler error if yotta configuration is not provided */
+#if !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_OFFSET)      || !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_NUMBER) || \
```

Comment:
This needs an additional `TARGET_HAS_NVIC` guard. This can be defined by the target and/or assumed automatically for Cortex-M3 and above (and of course, assumed as "off" by default for CM0)